alacritty/src/config/ui_config.rs

```diff
@@ -324,8 +325,13 @@ pub enum HintAction {
 /// Hint configuration.
 #[derive(Deserialize, Clone, Debug, PartialEq, Eq)]
 pub struct Hint {
-    /// Regex for finding matches.
-    pub regex: LazyRegex,
+    /// Apply this hint to regex matches.
+    #[serde(default)]
+    pub regex: Option<LazyRegex>,
+
+    /// Apply this hint to hyperlinks.
+    #[serde(default)]
+    pub hyperlinks: bool,
```

Comment:
By making both of these optional without any logic for checking that one of them is present, you're allowing users to specify hints with neither `regex` nor `hyperlinks`. I think that's probably the biggest functional issue with the PR in its current state.

If a config is invalid, we need to point that out. Especially because typos like `regxe` can be easily missed and users will start complaining that their stuff doesn't work. So you should ensure that at least one is present and throw an error otherwise.